cpp/src/io/parquet/page_decode.cuh

```diff
@@ -921,43 +931,46 @@ __device__ void gpuUpdateValidityOffsetsAndRowIndices(int32_t target_input_value
  * @param[in] target_leaf_count Target count of non-null leaf values to generate indices for
  * @param[in] rep Repetition level buffer
  * @param[in] def Definition level buffer
- * @param[in] t Thread index
+ * @param[in] warp Warp cooperative group
  * @tparam rolling_buf_size Size of the cyclic buffer used to store value data
  * @tparam level_t Type used to store decoded repetition and definition levels
  * @tparam state_buf Typename of the `state_buf` (usually inferred)
  */
 template <int rolling_buf_size, typename level_t, typename state_buf>
-__device__ void gpuDecodeLevels(page_state_s* s,
-                                state_buf* sb,
-                                int32_t target_leaf_count,
-                                level_t* const rep,
-                                level_t* const def,
-                                int t)
+__device__ void gpuDecodeLevels(
+  page_state_s* s,
+  state_buf* sb,
+  int32_t target_leaf_count,
+  level_t* const rep,
+  level_t* const def,
+  cg::thread_block_tile<cudf::detail::warp_size, cg::thread_block> const& warp)
```

Comment:
good change to stop using t here. looks like t was used with the assumption that `gpuDecodeLevels` is only run by the first warp.